numba/targets/boxing.py

```diff
@@ -580,7 +580,9 @@ def _python_list_to_native(typ, obj, c, size, listptr, errorptr):
                 list.parent = obj
             # Stuff meminfo pointer into the Python object for
             # later reuse.
-            c.pyapi.list_set_private_data(obj, list.meminfo)
+            with c.builder.if_then(c.builder.not_(c.builder.load(errorptr)),
+                                                  likely=False):
```

Comment:
Yes, as long as this is called only once per function argument (at most), there's probably not much potential for concrete gains.